### PR TITLE
refactor!: remove json property attributes

### PR DESCRIPTION
**Pull request by peterdeme**
Refactor:
- make serialization logic easier by auto applying snake case.
- some very small refactors

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     env:
       DOTNET_CLI_TELEMETRY_OPTOUT: "true"
+      DOTNET_SKIP_FIRST_TIME_EXPERIENCE: "true"
```
Comment (ferhatelmas) [inline .github/workflows/ci.yaml:15]: sounds like a good idea

File: src/Clients/MessageClient.cs [hunk 2]
```diff
@@ -6,6 +6,7 @@
 using Newtonsoft.Json;
```
Comment (ferhatelmas) [inline src/Clients/MessageClient.cs:6]: needed here?

File: src/Clients/StreamClientFactory.cs [hunk 3]
```diff
@@ -67,11 +67,12 @@ public StreamClientFactory(string apiKey, string apiSecret, Action<ClientOptions
             var opts = new ClientOptions();
             opts.OverrideWithEnvVars();
             clientOptionsConfigurer?.Invoke(opts);
+            opts.EnsureValid();
 
             var jwtGeneratorClient = new JwtGeneratorClient();
             var generatedJwt = jwtGeneratorClient.GenerateServerSideJwt(apiSecret);
             var assemblyVersion = typeof(StreamClientFactory).GetTypeInfo().Assembly.GetName().Version;
-            var sdkVersion = string.Join(".", assemblyVersion.Major, assemblyVersion.Minor, assemblyVersion.Build);
+            var sdkVersion = assemblyVersion.ToString(3);
```
Comment (ferhatelmas) [inline src/Clients/StreamClientFactory.cs:75]: nice touch

File: src/Models/AppSettings.cs [hunk 4]
```diff
@@ -50,91 +43,44 @@ public enum UniqueUsernameEnforcementPolicy
...
     {
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "apn_config")]
-        public APNConfig APNConfig { get; set; }
-
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "firebase_config")]
+        public APNConfig ApnConfig { get; set; }
         public FirebaseConfig FirebaseConfig { get; set; }
-
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "huawei_config")]
         public HuaweiConfig HuaweiConfig { get; set; }
```
Comment (ferhatelmas) [inline src/Models/AppSettings.cs:76]: himm, we're missing xiaomi config here

File: src/Models/AppSettings.cs [hunk 5]
```diff
@@ -146,55 +92,30 @@ public static class APNAuthType
...
-
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "p12_cert")]
+        [JsonProperty("p12_cert")]
         public string P12Cert { get; set; }
     }
 
     public class FirebaseConfig
     {
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "server_key")]
         public string ServerKey { get; set; }
```
Comment (ferhatelmas) [inline src/Models/AppSettings.cs:111]: we're missing `credentials_json` of v2

File: src/Models/ClientOptions.cs [hunk 6]
```diff
@@ -42,5 +43,11 @@ internal void OverrideWithEnvVars()
             if (!string.IsNullOrWhiteSpace(urlEnvVar))
                 BaseUrl = new Uri(urlEnvVar, UriKind.Absolute);
         }
+
+        internal void EnsureValid()
+        {
+            if (Timeout.TotalSeconds < 1 || Timeout.TotalSeconds > 60)
+                throw new InvalidClientOptionsException("Timeout must be between 1 and 60 seconds");
```
Comment (ferhatelmas) [inline src/Models/ClientOptions.cs:50]: 60 is probably a bad value, let's ensure max is 59

**Review by ferhatelmas — COMMENTED** [review]
Looks good, nice touches

Two points:
* we have missing configs but better to handle them separately
* some fields are renamed, ID -> Id this would be breaking change

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (peterdeme) [reply .github/workflows/ci.yaml:15]: 🙏

File: src/Clients/StreamClientFactory.cs (hunk 3 above)
Comment (peterdeme) [reply src/Clients/StreamClientFactory.cs:75]: 🙏

File: src/Models/AppSettings.cs (hunk 4 above)
Comment (peterdeme) [reply src/Models/AppSettings.cs:77]: added ✅

File: src/Models/AppSettings.cs (hunk 5 above)
Comment (peterdeme) [reply src/Models/AppSettings.cs:113]: added ✅

File: src/Models/ClientOptions.cs (hunk 6 above)
Comment (peterdeme) [reply src/Models/ClientOptions.cs:50]: modified ✅

File: src/Models/AppSettings.cs [hunk 7]
```diff
@@ -149,56 +94,39 @@ public static class APNAuthType
...
         public string Id { get; set; }
+        public string Secret { get; set; }
+    }
 
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "secret")]
+    public class XiaomiConfig
+    {
+        public string PackageName { get; set; }
         public string Secret { get; set; }
+        public bool Disabled { get; set; }
```
Comment (ferhatelmas) [inline src/Models/AppSettings.cs:129]: missing it in apn & huawei

File: src/Models/AppSettingsWithDetails.cs [hunk 8]
```diff
@@ -5,67 +5,40 @@ namespace StreamChat.Models
...
 
     public class PushNotificationFields
     {
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "apn")]
         public APNFields APN { get; set; }
-
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "firebase")]
         public FirebaseFields Firebase { get; set; }
-
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "huawei")]
```
Comment (ferhatelmas) [inline src/Models/AppSettingsWithDetails.cs:35]: missing xiaomi

**Review by ferhatelmas — DISMISSED** [review]
two small issues commented

also it would be really easier if you can push a new commit instead of push force, hard to see changed for big diffs

File: src/Models/AppSettings.cs (hunk 7 above)
Comment (peterdeme) [reply src/Models/AppSettings.cs]: added ✅

File: src/Models/AppSettingsWithDetails.cs (hunk 8 above)
Comment (peterdeme) [reply src/Models/AppSettingsWithDetails.cs:35]: ✅

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**